### PR TITLE
Multiple hardware fixes

### PR DESCRIPTION
**Pull request by digitalentity**
- Support for MS5607 baro. This is a cheaper alternative to MS5611. Unfortunately it is not possible to distinguish between the two when doing auto-detection. Datasheet doesn't specify anything specific to look for.
- Support for Spansion S25FL116 16Mbit SPI flash. Small footprint memory chip.
- Do HMC5883 boot-time calibration with 4Ga gain instead of 2.5Ga. Less likely saturation failures.

### CODE REVIEW

File: src/main/drivers/barometer_spi_ms56xx.c [hunk 1]
```diff
@@ -24,50 +24,50 @@
...
-#ifdef USE_BARO_SPI_MS5611
+#ifdef USE_BARO_SPI_MS56XX
 
-#define DISABLE_MS5611      IOHi(ms5611CsPin)
-#define ENABLE_MS5611       IOLo(ms5611CsPin)
+#define DISABLE_MS56XX      IOHi(ms56xxCsPin)
+#define ENABLE_MS56XX       IOLo(ms56xxCsPin)
 
-static IO_t ms5611CsPin = IO_NONE;
+static IO_t ms56xxCsPin = IO_NONE;
```
Comment (martinbudden) [inline src/main/drivers/barometer_spi_ms56xx.c]: Since this is `static` I'd just call it `spiCsnPin`

Comment (digitalentity) [reply src/main/drivers/barometer_spi_ms56xx.c]: Good point

File: src/main/drivers/barometer_spi_ms56xx.c [hunk 2]
```diff
@@ -24,50 +24,50 @@
...
-#include "barometer_ms5611.h"
+#include "barometer_ms56xx.h"
 
-#ifdef USE_BARO_SPI_MS5611
+#ifdef USE_BARO_SPI_MS56XX
 
-#define DISABLE_MS5611      IOHi(ms5611CsPin)
-#define ENABLE_MS5611       IOLo(ms5611CsPin)
+#define DISABLE_MS56XX      IOHi(ms56xxCsPin)
+#define ENABLE_MS56XX       IOLo(ms56xxCsPin)
```
Comment (martinbudden) [inline src/main/drivers/barometer_spi_ms56xx.c]: I'd parameterise these macros, eg `#define DISABLE_MS56XX(spiCsnPin)       IOHi(spiCsnPin)`

**Comment by martinbudden** [comment]
At some point we can put the SPI Instance in the new `busDevice_t` and that will eliminate a lot of duplicate SPI read and write functions.

**Comment by digitalentity** [comment]
Yes, we'll just have `busRead` and `busWrite`

**Pull request merged**